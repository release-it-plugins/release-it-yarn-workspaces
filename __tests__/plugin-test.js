const { createTempDir } = require('broccoli-test-helper');
const { factory, runTasks } = require('release-it/test/util');
const Plugin = require('../index');

const namespace = 'release-it-yarn-workspaces';

class TestPlugin extends Plugin {
  constructor() {
    super(...arguments);

    this.responses = {};

    this.commands = [];
    this.shell.execFormattedCommand = async (command, options) => {
      this.commands.push([command, options]);
      if (this.responses[command]) {
        return Promise.resolve(this.responses[command]);
      }
    };
  }
}

function buildPlugin(config = {}, _Plugin = TestPlugin) {
  const options = { [namespace]: config };
  const plugin = factory(_Plugin, { namespace, options });

  return plugin;
}

function json(obj) {
  return JSON.stringify(obj, null, 2);
}

describe('release-it-yarn-workspaces', () => {
  let ROOT = process.cwd();
  let dir;

  beforeEach(async () => {
    dir = await createTempDir();

    process.chdir(dir.path());
  });

  afterEach(async () => {
    process.chdir(ROOT);
    await dir.dispose();
  });

  it('it executes commands with defaults', async () => {
    let plugin = buildPlugin();

    dir.write({
      'package.json': json({
        name: 'root',
        version: '0.0.0',
        license: 'MIT',
        private: true,
        workspaces: ['packages/*'],
      }),
      packages: {
        foo: {
          'package.json': json({
            name: 'foo',
            version: '0.0.0',
            license: 'MIT',
          }),
        },
        bar: {
          'package.json': json({
            name: 'bar',
            version: '0.0.0',
            license: 'MIT',
          }),
        },
      },
    });

    await runTasks(plugin);

    expect(plugin.commands).toMatchSnapshot();
  });
});
