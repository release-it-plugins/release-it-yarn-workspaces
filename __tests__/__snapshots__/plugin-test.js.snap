// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`release-it-yarn-workspaces it executes commands with defaults 1`] = `
Array [
  Array [
    "npm version 1.0.1 --no-git-tag-version",
    Object {},
  ],
  Array [
    "npm version 1.0.1 --no-git-tag-version",
    Object {},
  ],
]
`;
